core/src/main/java/org/apache/calcite/prepare/CalcitePrepareImpl.java

```diff
@@ -937,8 +944,10 @@ public <R> R perform(CalciteServerStatement statement,
         prepareContext.getRootSchema().plus(), statement);
   }
 
-  /** Holds state for the process of preparing a SQL statement. */
-  static class CalcitePreparingStmt extends Prepare
+  /** Holds state for the process of preparing a SQL statement.
+   *  Overload this class and {@link #createSqlValidator} to supply custom validation logic.
+   * */
+  public static class CalcitePreparingStmt extends Prepare
```

Comment:
improve javadoc formatting. second sentence should be a new paragraph. use `**/` rather than `* */`.